### PR TITLE
🎉  added multicnv tools and wf

### PR DESCRIPTION
**Pull request by dmiller15**
## Description

This PR does 4 things:
- Imports the MultiCNV workflow/tools and brings them closely in line with our styling standards
- Adds functionality to prepend all outputs with an output_basename
- Expands cfree ploidy file recognition to include `.info.txt` files in addition to `.bam_info.txt` files
- Adds functionality to override the `case_id` and `sample_id` file metadata (can also be used in cases where there is no metadata)

Part of https://github.com/d3b-center/bixu-tracker/issues/1324

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] https://cavatica.sbgenomics.com/u/brownm28/mb-controlfreec-troubleshoot/tasks/80be664f-2bc0-49e6-b309-feb5ae9ff66b/

**Test Configuration**:
* Environment: cwltool and cavatica
* Test files: see test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have committed any related changes to the PR

### CODE REVIEW

File: tools/sb_multicnv_rename_files.cwl [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+
+                  // Ploidy files extensions
+
+                  // Sclust
+                  if (path.endsWith('.out_cn_summary.txt')){
+                      var caller = 'sclust'
+                      var ext = '.out_cn_summary.txt'
+                      var ploidy = 'ploidy'
+                  }
+                  // Control FREEC
```
Comment (migbro) [inline tools/sb_multicnv_rename_files.cwl:103]: nice!

File: workflow/sb_multicnv_wf.cwl [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+  - **Facets** ('.cncf.tsv')
+  - **SBG Conseca CNV** ('.conseca.tsv')
+  - **Sclust** ('.allelic_states.txt'). 
+
+
+
+  Files with information about the ploidy value can be used only for specific callers:  
+    
+  - **PURPLE** ('.purple.purity.tsv')
+  - **ControlFREEC** ('.bam_info.txt')
```
Comment (migbro) [inline workflow/sb_multicnv_wf.cwl:54]: Update this to say `.info.txt` OR `.bam_info.txt` or not worth it?

File: workflow/sb_multicnv_wf.cwl [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+    label: Result after segmentation
+    doc: |-
+      Resulted file after merging the regions with the same final status excluding 'ambiguous' regions'
+    sbg:fileTypes: CSV
+  segment_results_with_ambiguous:
+    type: File[]?
+    outputSource: add_basename_segment_results_with_ambiguous/output
+    label: Result after segmentation with ambiguous
+    doc: Resulted file after merging the regions with the same final status
+    sbg:fileTypes: CSV
```
Comment (migbro) [inline workflow/sb_multicnv_wf.cwl:199]: Hmmm, does `sbg:fileTypes` need to be quoted to work with sbpack?

**Review by migbro — APPROVED** [review]
Seems good, just minor comments/questions

**Review by sickler-alex — APPROVED** [review]

File: workflow/sb_multicnv_wf.cwl (hunk 2 above)
Comment (dmiller15) [reply workflow/sb_multicnv_wf.cwl:54]: Will do. Should be quick.

File: workflow/sb_multicnv_wf.cwl (hunk 3 above)
Comment (dmiller15) [reply workflow/sb_multicnv_wf.cwl:199]: Weirdly enough, it worked fine. Must be more forgiving when it is in the yaml format.

Comment (migbro) [reply workflow/sb_multicnv_wf.cwl:199]: ooooh, interesting

**Pull request merged**